test/features/showandhide/readonlyFieldsComplex.feature

```diff
@@ -0,0 +1,24 @@
+@functional @conditional @showandhide
+Feature: Set of scenarios testing show & hide functionality and conditional logic
+
+  Background:
+    Given the following definition for 'complex case fields'
+      | ID       | CaseEvent    | Path                    | Display Context | Field show condition                   |
+      | MySchool | createSchool | Class.ClassMandatoryFor | MANDATORY       |                                        |
+      | MySchool | updateSchool | Class.ClassMandatoryFor | READONLY        | Name="alwaysHideThisField"             |
+      | MySchool | updateSchool | Class.ClassName         | OPTIONAL        | MySchool.Class.ClassMandatoryFor="MSc" |
+    Given I have logged in
+
+  Scenario: shows a field when Show Condition uses a readonly nested in complex field value from a previous event
+    Given a case type containing Complex and Collection types exists
+    When I populate the form with the school data with a degree field set to 'MSc'
+    And I move forward 2 pages
```

Comment:
to be more descriptive we should be stating that we are creating/submitting the case rather than just saying move forward 2 pages (which I assume would be hitting the continue thens submit button and thus creating/submitting the case)